tests/components/zha/test_config_flow.py

```diff
@@ -1,125 +1,237 @@
 
-    assert result["type"] == "abort"
+async def test_user_flow_show_form(hass):
+    """Test user step form."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={CONF_SOURCE: SOURCE_USER},
+    )
+
+    assert result["type"] == RESULT_TYPE_FORM
+    assert result["step_id"] == "user"
 
 
-async def test_import_flow(hass):
-    """Test import from configuration.yaml ."""
-    flow = config_flow.ZhaFlowHandler()
-    flow.hass = hass
+async def test_user_flow_manual(hass):
+    """Test user flow manual entry."""
 
-    result = await flow.async_step_import(
-        {"usb_path": "/dev/ttyUSB1", "radio_type": "xbee"}
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={CONF_SOURCE: SOURCE_USER},
+        data={zigpy.config.CONF_DEVICE_PATH: config_flow.CONF_MANUAL_PATH},
     )
+    assert result["type"] == RESULT_TYPE_FORM
+    assert result["step_id"] == "pick_radio"
 
-    assert result["type"] == "create_entry"
-    assert result["title"] == "/dev/ttyUSB1"
-    assert result["data"] == {"usb_path": "/dev/ttyUSB1", "radio_type": "xbee"}
 
+async def test_pick_radio_flow(hass):
+    """Test radio picker."""
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={CONF_SOURCE: "pick_radio"}, data={CONF_RADIO_TYPE: "ezsp"}
+    )
+    assert result["type"] == RESULT_TYPE_FORM
+    assert result["step_id"] == "port_config"
 
-async def test_import_flow_existing_config_entry(hass):
-    """Test import from configuration.yaml ."""
-    MockConfigEntry(domain=DOMAIN, data={"usb_path": "/dev/ttyUSB1"}).add_to_hass(hass)
-    flow = config_flow.ZhaFlowHandler()
-    flow.hass = hass
 
-    result = await flow.async_step_import(
-        {"usb_path": "/dev/ttyUSB1", "radio_type": "xbee"}
+async def test_user_flow_existing_config_entry(hass):
+    """Test if config entry already exists."""
+    MockConfigEntry(domain=DOMAIN, data={"usb_path": "/dev/ttyUSB1"}).add_to_hass(hass)
+    await setup.async_setup_component(hass, "persistent_notification", {})
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={CONF_SOURCE: SOURCE_USER}
     )
 
     assert result["type"] == "abort"
 
 
-async def test_check_zigpy_connection():
-    """Test config flow validator."""
+async def test_probe_radios(hass):
+    """Test detect radios."""
+    app_ctrl_cls = MagicMock()
+    app_ctrl_cls.SCHEMA_DEVICE = zigpy.config.SCHEMA_DEVICE
+    app_ctrl_cls.probe = AsyncMock(side_effect=(True, False))
 
-    mock_radio = tests.async_mock.MagicMock()
-    mock_radio.connect = tests.async_mock.AsyncMock()
-    radio_cls = tests.async_mock.MagicMock(return_value=mock_radio)
+    with patch.dict(config_flow.RADIO_TYPES, {"ezsp": {CONTROLLER: app_ctrl_cls}}):
+        res = await config_flow.detect_radios("/dev/null")
+        assert app_ctrl_cls.probe.await_count == 1
+        assert res[CONF_RADIO_TYPE] == "ezsp"
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5462445)